### PR TITLE
add chat page

### PR DESCRIPTION
**Pull request by ziyangcS2023**

### CODE REVIEW

**Comment by ziyangcS2023** [comment]
chatting feature done

File: src/pages/chat.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            <Heading level={1} align="center" className="mb-2">
+                {name}
+            </Heading>
+            {messages.map((message) => (
+                <ChatMessage
+                    msg={message.msg}
+                    isSender={message.isSender}
+                ></ChatMessage>
+            ))}
+            <Input
```
Comment (congruiyS2023) [inline src/pages/chat.jsx]: Please use [`CustomInputWithSubmit`](https://github.com/congruiyS2023/platemate-web/blob/54da89ae5b2ef00b4ced187d4aeb087837acaacb/src/components/CustomInputs.jsx#L34) here.

**Review by congruiyS2023 — CHANGES_REQUESTED** [review]
After click on the send button, my text input in not cleared, neither do I see my sent message on screen.

File: src/pages/chat.jsx (hunk 1 above)
Comment (ziyangcS2023) [reply src/pages/chat.jsx]: resolved in [78f1128](https://github.com/congruiyS2023/platemate-web/pull/7/commits/78f11287f98a26d5d9946513ac052e731bb4e512)

**Review by congruiyS2023 — CHANGES_REQUESTED** [review]
Great!! Now just change the style of ChatMessage to match the style guide then I think you can begin to incorporate this page to @YinzhuSU 's user flow.
Chat

**Comment by yinzhusS2023** [comment]
indeed!

**Comment by ziyangcS2023** [comment]
fixed stylings on [3f39b03](https://github.com/congruiyS2023/platemate-web/pull/7/commits/3f39b036ce548901d9d593162c9b3148f507959a)

**Comment by congruiyS2023** [comment]
Lgtm! @YinzhuSU Please incorporate the route name!

**Comment by yinzhusS2023** [comment]
Hi @congruiyS2023 

I checked the PR, it looks good.
I also resolved the conflict to the master branch. Would you please resolve your Change request and merge it to master?
Thanks!

**Review by yinzhusS2023 — APPROVED** [review]
Nice work!
Next time before you push your code to remote, remember to pull the latest code from master, merge the master to your feature branch and resolve any conflicts, then push and make the PR.

File: src/App.js [hunk 2]
```diff
@@ -32,6 +33,7 @@ function App() {
               <Route path="/reward/coupons" element={<Coupons />} />
               <Route path="/user-home/join-challenge" element={<JoinChallange />} />
               <Route path="/user-home/join-challange-success" element={<JoinChallangeSuccess />} />
+              <Route path="/chat" element={<Chat />} />
```
Comment (congruiyS2023) [inline src/App.js]: @ziyangcS2023 @YinzhuSU Please change the path to `/recycle/chat` to match the change in route names, thanks!

**Review by congruiyS2023 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 3]
```diff
@@ -32,6 +33,7 @@ function App() {
               <Route path="/reward/coupons" element={<Coupons />} />
               <Route path="/user-home/join-challenge" element={<JoinChallange />} />
               <Route path="/user-home/join-challange-success" element={<JoinChallangeSuccess />} />
+              <Route path="/recycle/chat-with-recycle-companies" element={<Chat />} />
```
Comment (congruiyS2023) [inline src/App.js:36]: If we are really aiming for full REST, the path should be `recycle/chat/:recycleCompanyId`, but I think we're good here. Nice work guys!

**Review by congruiyS2023 — APPROVED** [review]
Nice work guys!

File: src/App.js (hunk 3 above)
Comment (congruiyS2023) [reply src/App.js:36]: @ziyangcS2023 @YinzhuSU Just FYI ⬆️ I'm resolving it rn.

**Pull request merged**